shared/constants.js

```diff
@@ -25,92 +25,233 @@
             appTypesToPath: {
                 'native': 'AndroidNativeTemplate',
                 'native_kotlin': 'AndroidNativeKotlinTemplate'
-            }
+            },
+            commands: ['create', 'createWithTemplate', 'version']            
         },
         forcehybrid: {
             name: 'forcehybrid',
-            description: 'command line utility for building hybrid mobile applications using Salesforce Mobile SDK',
+            description: 'tool for building hybrid mobile applications using Salesforce Mobile SDK',
+            sfdx_topic: 'hybrid',
+            sfdx_description: 'command for building an hybrid mobile application using Saleforce Mobile SDK',
             dir: 'hybrid',
             platforms: ['ios', 'android'],
             toolNames: ['git', 'npm', 'cordova'],
             appTypes: ['hybrid_local', 'hybrid_remote'],
             appTypesToPath: {
                 'hybrid_local': 'HybridLocalTemplate',
                 'hybrid_remote': 'HybridRemoteTemplate'
-            }
+            },
+            commands: ['create', 'createWithTemplate', 'version']            
         },
         forcereact: {
             name: 'forcereact',
-            description: 'command line utility for building react native mobile applications using Salesforce Mobile SDK',
+            description: 'tool for building react native mobile applications using Salesforce Mobile SDK',
+            sfdx_topic: 'reactnative',
+            sfdx_description: 'command for building a React Native mobile application using Saleforce Mobile SDK',
             dir: 'react',
             platforms: ['ios', 'android'],
             toolNames: ['git', 'npm', 'pod'],
             appTypes: ['react_native'],
             appTypesToPath: {
                 'react_native': 'ReactNativeTemplate'
-            }
+            },
+            commands: ['create', 'createWithTemplate', 'version']
         }
     },
 
-    tools: {
-        git: {
-            checkCmd: 'git --version',
-            minVersion: '2.13'
+    args: {
+        platform: {
+            name: 'platform',
+            'char': 'p',
+            description: cli => 'Comma separated platforms (' + cli.platforms.join(', ') + ')',
+            prompt: cli => 'Enter the target platform(s) separated by commas (' + cli.platforms.join(', ') + '):',
+            error: cli => val => 'Platform(s) must be in ' + cli.platforms.join(', '),
+            validate: cli => val => !val.split(",").some(p=>cli.platforms.indexOf(p) == -1)
         },
-        npm: {
-            checkCmd: 'npm -v',
-            minVersion: '3.10'
+        appType: {
+            name:'apptype',
+            'char':'t',
+            description: cli => 'Application Type (' + cli.appTypes.join(', ') + ')',
+            prompt: cli => 'Enter your application type (' + cli.appTypes.join(', ') + '):',
+            error: cli => val => 'App type must be ' + cli.appTypes.join(' or ') + '.',
```

Comment:
error and validate are function of the chosen cli and the value.
Instead of defining them as function of (cli, val), we have them as a function of (cli) that returns a function of (val).